misaligned.c

```diff
@@ -1,21 +1,56 @@
 #include <stdio.h>
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
+typedef struct
+{
+    int pairNumber;
+    const char * majorColor;
+    const char * minorColor;
+} colorMapStruct;
+
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+colorMapStruct ColorMaps[30];
+
+
+int GetColorMap(colorMapStruct * colorMaps)
+{
+    int i = 0, j = 0, index = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            colorMaps[index].pairNumber = i * 5 + j;
+            colorMaps[index].majorColor = majorColor[i];
+            colorMaps[index].minorColor = minorColor[i];
+            ++index;
         }
     }
     return i * j;
 }
 
-int main() {
-    int result = printColorMap();
-    assert(result == 25);
+void printColorMap()
+{
+    int numCombinations = GetColorMap(ColorMaps);
+    for (int index = 0; index < numCombinations; ++index)
+    {
+        printf("%d | %s | %s\n", ColorMaps[index].pairNumber, ColorMaps[index].majorColor, ColorMaps[index].minorColor);
+    }
+}
+
+void testColorMap(void)
+{
+    int numCombinations = GetColorMap(ColorMaps);
+    for (int index = 0, pairNumber = 1; index < numCombinations; ++index, ++pairNumber)
+    {
+        assert(pairNumber == ColorMaps[index].pairNumber);

```

Comment:
if numCombinations value = 10, will majorColor[10] execute properly?

Reply:
yes, it will lead to out of bound. it is corrected now. @prithamvshetty

Reply:
Important point is, do you have a test case that could have caught this error?
If not, then test cases are not sufficient 😊

Reply:
yes, you are correct. I did add a test case in line 49. Even changed the accessibility of majorColor[], minorColor[] within its bounds.
-> (index % MAX_MAJOR_COLORS)
-> (index % MAX_MINOR_COLORS)
Maybe i will also check if there are still any missing test cases. @prithamvshetty